pandas/tests/api/test_api.py

```diff
@@ -237,8 +245,14 @@ def test_depr(self):
 
 
 class TestApi(Base):
-    allowed = ["types", "extensions", "indexers", "interchange", "typing"]
-    allowed_typing = [
+    allowed_api_dirs: list[str] = [
```

Comment:
By-and-large, we do not type-hint test code. I think we should keep the status-quo here unless an explicit decision is made to add type-hints. Can you remove throughout.